### PR TITLE
nbMontage

### PR DESCRIPTION
**Pull request by qtcwt**
Most changes are directly copied from the private nbMontage repo. 
Tested. Should work. Extra manual checks are needed.

### CODE REVIEW

**Comment by qtcwt** [comment]
TODO:

- [x] Update scripts so they match the new indexing 
- [x] Update README

File: Makefile [hunk 1]
```diff
@@ -162,12 +162,28 @@ CXXFLAGS:= -fopenmp -pthread -g -fpic $(WARNING_FLAGS) -D_REENTRANT -fno-strict-
 # define these build configurations).
 # To run a build, e.g. release, you would invoke:
 # make release
-BUILDS :=release debug ngc release32 debug32 mnemosyne pronto-full pronto-sync graph-rec
+BUILDS :=release debug ngc release32 debug32 mnemosyne pronto-full pronto-sync graph-rec tsx vread
```
Comment (roghnin) [inline Makefile:165]: Do we still want to live tsx label here?

File: ext/README.html [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (roghnin) [inline ext/README.html]: Is this file auto-generated? Seems likely to me. We can just keep the .md file if possible.

**Review by roghnin — COMMENTED** [review]
I've made a pass through all the changes. Most of them are verified in private branch so I didn't bother testing them again.

One issue with compilation: mtm_i.h:477-480 from Mnemosyne are not declared as external variables, so linking may fail due to duplicated definitions. I'll push a fix after some testing.

I'm also updating execution scripts based on comments below.

File: Makefile (hunk 1 above)
Comment (qtcwt) [reply Makefile:165]: Let's just keep this. I have a USE_TSX path in CAS_verify.

File: ext/README.html (hunk 2 above)
Comment (qtcwt) [reply ext/README.html]: I didn't add this intentionally, so yes this should be somehow auto-generated. Deleted.

**Comment by qtcwt** [comment]
Alright. I've updated all scripts and README. This PR should be ready to merge. @roghnin

**Comment by roghnin** [comment]
> Alright. I've updated all scripts and README. This PR should be ready to merge. @roghnin

Cool. Thanks a lot!
I've added a check for test and rideable name duplications. Merging.

**Pull request merged**